src/server/api/routers/scores.ts

```diff
@@ -128,73 +150,57 @@ export const scoresRouter = createTRPCRouter({
         scope: "scores:CUD",
       });
 
-      return ctx.prisma.score.create({
-        data: {
-          trace: {
-            connect: {
-              id: trace.id,
-            },
+      // Delete all scores that are not in the input
+      // User removed them in the UI
+      await ctx.prisma.score.deleteMany({
```

Comment:
I think this should happen in a transaction together with all the upserts below.